src/consensus/params.cpp

```diff
@@ -8,12 +8,30 @@
 #include <key_io.h>
 #include <script/standard.h>
 #include "upgrades.h"
+#include "util.h"
 
 namespace Consensus {
     bool Params::NetworkUpgradeActive(int nHeight, Consensus::UpgradeIndex idx) const {
         return NetworkUpgradeState(nHeight, *this, idx) == UPGRADE_ACTIVE;
     }
 
+    bool Params::FeatureActive(const int nHeight, const Consensus::ConsensusFeature feature) const {
+        auto requires = WhatDependsOn(feature);
+        if (vRequiredFeatures.count(feature) > 0 ||
+            Any(requires, [&](Consensus::ConsensusFeature feat) { return FeatureActive(nHeight, feat); })) {
```

Comment:
This would recurse infinitely if there were a cycle in the dependency graph. That's fine since it is [documented](https://github.com/zcash/zcash/pull/4480/files#diff-f5aa51ec54f17eba17214e33d06708d02f073dc9edaa271e05787b43d21a3b73R155) that the graph must be acyclic, and any such hangs are likely to be deterministic (the first time a feature is tested for).